cucumber-selenium-tests/src/test/java/com/skillstorm/StepDefinitions/GenericStepDefinitions.java

```diff
@@ -1,19 +1,88 @@
 package com.skillstorm.StepDefinitions;
 
+import io.cucumber.java.en.*;
 
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.Select;
+import org.openqa.selenium.support.ui.WebDriverWait;
 
-public class GenericStepDefinitions {
+import com.skillstorm.Utilities.Navigator;
 
-    // @Given("an example scenario")
-    // public void anExampleScenario() {
-    // }
 
-    // @When("all step definitions are implemented")
-    // public void allStepDefinitionsAreImplemented() {
-    // }
+//This class implements some step definitions that are commonly used among our tests. You must pass in the web driver you're using as a parameter.
+public class GenericStepDefinitions{
 
-    // @Then("the scenario passes")
-    // public void theScenarioPasses() {
-    // }
+    WebDriver driver;
+    WebDriverWait wait;
+    Navigator navigator = new Navigator(driver);
```

Comment:
will this try to assign the variaable "driver" to navigator before it gets initiated?
should the  "= new Navigator(driver);" part of this code be in the the @BeforeSuite or @BeforeClass definition?

Reply:
I suggest having anything "new" (instantiated) in a public void before() method, attached with @Before